docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-cloud-providers/azure.md

```diff
@@ -110,6 +124,389 @@ Only hosts expected to be load balancer back ends need to be in this group.
 
     </details>
 
-1. Under the **Cluster Configuration > Advanced** section, click **Add** under **Additional Controller Manager Args** and add this flag: `--configure-cloud-routes=false`
+3. Under the **Cluster Configuration > Advanced** section, click **Add** under **Additional Controller Manager Args** and add this flag: `--configure-cloud-routes=false`
+
+4. Click the **Create** button to submit the form and create the cluster.
+
+## Cloud Provider Configuration
+
+Rancher will automatically create a new Network Security Group, Resource Group, Availability Set, Subnet, and Virtual Network. If you already have some or all of these created, you will need to specify them before creating the cluster. You can check RKE1 Node Templates for or RKE2 Machine Pools to see more of these automatically generated names and update them if necessary.
+
+**Refer to the full list of configuration options in the [upstream docs](https://cloud-provider-azure.sigs.k8s.io/install/configs/).**
+
+:::note 
+1. `useInstanceMetadata` must be set to true for `providerID` to be configured correctly by the cloud provider.
+2. `excludeMasterFromStandardLB` must be set to false if nodes labeled `node-role.kubernetes.io/master` are required to be added to the backends of Azure Load Balancer (ALB).
+3. `loadBalancerSku` can be `basic` or `standard`. Basic SKU will be deprecated in September 2025. Refer to the [Azure upstream docs](https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-basic-upgrade-guidance#basic-sku-vs-standard-sku) for more information.
+:::
+
+Azure cloud provider supports reading the cloud config from Kubernetes secrets. The secret is a serialized version of azure.json file. When the secret is changed, the cloud controller manager will re-constructing itself without restarting the pod. It is recommended for the helm chart to read the Cloud Provider Config from secret.
+
+Note that the chart reads the Cloud Provider Config from a given secret name in the `kube-system` namespace. Since Azure cloud provider would read Kubernetes secrets, RBAC also needs to be configured. An example secret for the Cloud Provider Config is shown below, modify it as needed and create the secret.
+
+  ```yaml
+# azure-cloud-config.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: azure-cloud-config
+  namespace: kube-system
+type: Opaque
+stringData:
+  cloud-config: |-
+    {
+      "cloud": "AzurePublicCloud",
+      "tenantId": "<tenant-id>",
+      "subscriptionId": "<subscription-id>",
+      "aadClientId": "<client-id>",
+      "aadClientSecret": "<tenant-id>",
+      "resourceGroup": "docker-machine-ks",
+      "location": "westus",
+      "subnetName": "docker-machine-ks",  
+      "securityGroupName": "rancher-managed-kqmtsjgJ",
```

Comment:
securityGroupName should be same everywhere while creating a cluster. User has to provider securityGroupName at least 3 times while creating a cluster. Could you please mention that in documentation?

Reply:
@vivek-shilimkar Did you have to provide the securityGroupName for the worker machine pool as well?

Reply:
Yes, as per the available [guide](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-cloud-providers/azure#rke2-cluster-set-up-in-rancher) the Network Security Group should be same for all the pools and cloud provider configuration. If there is miss match, user gets an error that is similar to - `Security group not found`.